### DISCUSSION TITLE
3rd Party dependency models in our source code(i.e in Parent Model)  are not getting created under openapi components.schemas and not getting referenced from parent  as well.

### DISCUSSION THREAD

**Discussion by shahbaaz31here** [eclipse/microprofile-open-api#595] [Q&A]
We are using using openapi 3 - (org.eclipse.microprofile.openapi) 
**Source Code Parent Model** : oracle.apps.crm.service.workflow.dsl.model.WorkflowDslPayload
**Referenced models  for which source code is not in current project(Its coming from a dependency)** : io.serverlessworkflow.api.states.OperationState
Model(WorkflowDslPayload) referencing List<OperationState>:

```
@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class WorkflowDslPayload {
    @Schema(name = "states",type = SchemaType.ARRAY, description = "DSL State",implementation = OperationState.class)
    private List<OperationState> states = new ArrayList<>();
```
    
The /openAPI response only contains WorkflowDslPayload schema and it is not able to reference  other objects for which source **code is not present.**

**Open API snippet :**
```
tags:
- 
  name: WorkflowDSL
components: 
  schemas:
    WorkflowDslPayload: 
      properties:
        states: 
          description: DSL State
          items: 
            type: object
          type: array
info: 
  description: WorkflowDSL Microservice .....
  title: WorkflowDSL Microservice
  version: '1.0'
openapi: 3.0.3
.....
```

In the above Schema, states.items should point to $ref: '#/components/schemas/OperationState' but **it isn't happening.**
**Annotation on Controller :**
```
@Path("/")
    @POST
    @Operation(summary = "Create Workflows",
            description = "Create Workflows")
    @RequestBody(
            name = "Workflow DSL payload",
            description = "Payload for creating workflow DSL",
            content = @Content(
                    mediaType = "application/json",
                    schema = @Schema(implementation = WorkflowDslPayload.class)

            )
    )

    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    public Response defineWorkflow(@Valid WorkflowDslPayload dslRequest){...}
```

**Comment by MikeEdgar** [comment]
Hi @shahbaaz31here , this looks like an issue with the implementation and not the MicroProfile OpenAPI spec itself. What server/runtime are you using?

**Reply by shahbaaz31here** [reply]
I am using helidon https://helidon.io/docs/v3/mp/openapi/openapi. Does eclipse microprofile-open-api  by default  scans the models present in dependent jars (source code is not present) and adds them to components.schemas ? I was trying to apply the configuration  `mp.openapi.schema.` to include the class in schemas by referring https://download.eclipse.org/microprofile/microprofile-open-api-3.0/microprofile-openapi-spec-3.0.html#configuration  but somehow this is not working for me. 
```

mp.openapi.schema.io.serverlessworkflow.api.states.OperationState = { \
  "name": "states" \
  "type": "object", \
  "description": "Milliseconds since January 1, 1970, 00:00:00 GMT" \
}

```

**Reply by MikeEdgar** [reply]
This is either an issue in Helidon itself or smallrye-open-api used by Helidon. 

@tjquinno , FYI. I am guessing this is something to do with the Jandex constructed in Helidon prior to invoking the annotation scanning.

**Reply by tjquinno** [reply]
Yes, thanks @MikeEdgar. This user posted the same question in our support channel and I've responded there. Apologies for the noise here.
